Deprecate CONTAINER_TEMPTO_PROFILE_CONFIG

kokosing opened the pull request:
Deprecate CONTAINER_TEMPTO_PROFILE_CONFIG

Praveen2112 commented on testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentContainers.java:
```diff
@@ -34,4 +38,23 @@ public static boolean isPrestoContainer(String name)
+    /**
+     * Use this method only when you place `tempto-configuration.yaml' in environment configuration directory.
+     */
+    public static void configureTempto(Environment.Builder builder, DockerFiles.ResourceProvider configDir)
```
static import for `ResourceProvider`

Praveen2112 commented on testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentContainers.java:
```diff
@@ -34,4 +38,23 @@ public static boolean isPrestoContainer(String name)
+    public static void configureTempto(Environment.Builder builder, DockerFiles.ResourceProvider configDir)
+    {
+        builder.configureContainer(TESTS, dockerContainer -> {
+            String environmentName = configDir.getPath("..").toFile().getName();
```
Instead of deriving it, can we pass it as an argument.

kokosing replied:
I was done this way after: https://github.com/trinodb/trino/pull/8627#discussion_r700156415

> this filedescription is nice, but it's not copy-paste-friendly. would you consider dropping it?

Praveen2112 commented on testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/env/EnvironmentContainers.java:
```diff
@@ -34,4 +38,23 @@ public static boolean isPrestoContainer(String name)
+    public static void configureTempto(Environment.Builder builder, DockerFiles.ResourceProvider configDir)
+    {
+        builder.configureContainer(TESTS, dockerContainer -> {
+            String environmentName = configDir.getPath("..").toFile().getName();
```
Can we try something like this 
`configDir.getPath("tempto-configuration.yaml").getParent().toFile().getName()`

kokosing replied:
What is the difference?

findepi commented:
@kokosing i think i reviewed it in some other PR, right? has anything changed?

kokosing commented:
> has anything changed?

@findepi I addressed your comments. Remove a parameter that now @Praveen2112 wants to restore. And used that method for Kafka envs.

kokosing commented:
I extracted this from this https://github.com/trinodb/trino/pull/8627, as there I had more changes and it was causing issues with TLS environment.

findepi commented:
> > has anything changed?
> 
> @findepi I addressed your comments.

LGTM then, thanks!
